pkg/apis/softwarecomposition/v1beta1/network_types.go

```diff
@@ -0,0 +1,67 @@
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type Protocol string
+type CommunicationType string
+
+const (
+	ProtocolTCP  Protocol = "TCP"
+	ProtocolUDP  Protocol = "UDP"
+	ProtocolSCTP Protocol = "SCTP"
+
+	CommunicationTypeIngress CommunicationType = "internal"
+	CommunicationTypeEgress  CommunicationType = "external"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NetworkNeighborsList is a list of NetworkNeighbors.
+type NetworkNeighborsList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Items []NetworkNeighbors `json:"items"`
+}
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NetworkNeighbors represents a list of network communications for a specific workload.
+type NetworkNeighbors struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec NetworkNeighborsSpec `json:"spec"`
+}
+
+type NetworkNeighborsSpec struct {
+	*metav1.LabelSelector
```

Comment:
issue: missing the JSON struct tag

In order to properly embed a type into a JSON type definition, you need to provide the `json:",inline"` struct tag. Then the fields will be properly propagated to the embedding type.


```suggestion
	*metav1.LabelSelector `json:",inline"`
```

Reply:
And why embed it as a pointer for the `*metav1.LabelSelector`? I think this should be:

```suggestion
	metav1.LabelSelector `json:",inline"`
```